lib/Expression/Ast/VariableNode.php

```diff
@@ -2,7 +2,7 @@
 
 namespace PhpBench\Expression\Ast;
 
-class VariableNode implements Node
+final class VariableNode extends Node
```

Comment:
use inheritence with the AST nodes - this helps with template resolution, but also would have been a better idea from the start.